CAIPs/caip-211.md

```diff
@@ -0,0 +1,275 @@
+"default" or universal endpoints and definitions, which can be hard to anchor in
+explicit specifications unless a namespace profile of this CAIP has been
+published. (This may be unnecessary in the case of younger namespaces without
+customization built in at the per-chain or per-dapp layer). An empty array of
+`rpcEndpoints` or `rpcDocuments` sent in either direction should be interpreted
+differently than the absence of the array.  Namely, either array being present
+but empty in a request signals that a caller is requesting that custom endpoints
+or  definitions NOT be considered in CAIP-25 authorizations. Present but empty
+arrays in a response confirms this behavior, while its absence can be taken to
+mean the wallet either does not support custom endpoints/definitions or the user
+has opted out of that degree of trust, which for privacy/fingerprinting reasons,
+should not be handled by distinct codepaths (See Privacy Considerations).
+
+## Specification
+
+Two properties are added to the scope objects requested in `optionalScopes` and
+optionally returned in `sessionScopes`.  These are both [ordered] strings of arrays:
+
+1. `rpcEndpoints` is an array of zero or more URLs of RPC endpoints that
+   the caller would prefer the respondent to use, ordered by preference. Each
+   must be a valid URL that addresses an RPC endpoint. The respondent may
+   return it empty, reordered, with less, the same, or even more conformant URLs
+   than received.
+2. `rpcDocuments` is an array of zero or more URLs of machine-readable RPC
+   documents that the caller would prefer the respondent to use, ordered by
+   preference. This set of document collectively defines at least syntactically
+   if not also semantically any methods and/or notifications authorized by the
+   CAIP-25 authorization, in DESCENDING heirarchical authority. (For example,
+   any methods or notifications defined differently by multiple authorites will
+   be interpreted by whichever authority is closer to the 0-index of the array).
+   Each must be a valid URL that addresses a valid openRPC document. The
+   respondent may return it empty, reordered, with less, the same, or even more
+   conformant URLs than received.
+
+### Request
+
+A CAIP-211 request is a valid CAIP-25 except for the two additional properties. 
+
+Example:
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "provider_authorization",
+  "params": {
+    "requiredScopes": {
+      "eip155": {
+        "chains": ["eip155:1", "eip155:137"],
+        "methods": ["eth_sendTransaction", "eth_signTransaction", "eth_sign", "get_balance", "personal_sign"],
+        "notifications": ["accountsChanged", "chainChanged"]
+      },
+      "eip155:42069": {
+        "methods": ["get_balance", "chainChanged", "42069_sEcReTbAlAnCe"],
+        "rpcDocuments": ["https://openrpc.42069-chain.org/", "https://ethereum.github.io/execution-apis/api-documentation/"],
```

Comment:
this would not point to a documentation site, which is what https://ethereum.github.io/execution-apis/api-documentation/ is.

it would point to the actual openrpc document which is located here: https://raw.githubusercontent.com/ethereum/execution-apis/assembled-spec/refs-openrpc.json

Reply:
good catch!  curious that `https://ethereum.github.io/execution-apis/assembled-spec/refs-openrpc.json` doesn't work-- would it make sense to configure the gitpages to publish to, e.g., docs.ethereum.org so that the link would survive a migration off of (Microsoft-owned) infra?

(I'll update it with the accurate raw.githubusercontent.com URL for now regardless)